### PR TITLE
Create-Retrieve check in entries from native layer

### PR DESCRIPTION
**Pull request by aledustet**
Why:
----

One of the two data models that will drive the my health screen are check-ins. These should be made once per day and will hold just a check-in status of `Feeling well` or `Feeling not well`. As a user I should be able to go to the My Health screen, add my check-in for the day and see it reflected both on th `Today` tab and the `OverTime` tab. This data should persist from app session to session.

This Commit:
----

- Display todays check in status on today view
- Add todays check in from Today screen
- Update ios native module for checkin/symptom entry changes.
- Store and retrieve check-in entries from the native side
- Rename methods on symptoms helper

Caveat:
----

The data collection and creation only works on iOS at the moment since the updates to the native layer on android are pending.

Co-authored-by: Matt Buckley <matt.buckley@rocketinsights.com>
Co-authored-by: Devin Jameson <devin@thoughtbot.com>

save-check-in

### CODE REVIEW

File: src/MyHealth/SymptomLogContext.tsx [hunk 1]
```diff
@@ -8,50 +8,20 @@ import React, {
...
-      date: 1600804626042,
-      status: CheckInStatus.FeelingGood,
-    },
-    {
-      date: 1599919200000,
-      status: CheckInStatus.FeelingNotWell,
-    },
-  ])
-}
+import { getLogEntries, getCheckIns, addCheckIn } from "../gaen/nativeModule"
```
Comment (johnschoeman) [inline src/MyHealth/SymptomLogContext.tsx:16]:
```suggestion
import { getLogEntries, getCheckIns, addCheckIn } from "./nativeModule"
```

what do you think about creating a new NativeModule scoped to the MyHealth feature? this doesnt seem directly related to GAEN.

**Review by johnschoeman — APPROVED** [review]
Looks great to me!

File: src/MyHealth/SymptomLogContext.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+      const AddCheckInStatus = () => {
+        const { todaysCheckIn, addTodaysCheckIn } = useSymptomLogContext()
+
+        return (
+          <>
+            <Text>{todaysCheckIn.status}</Text>
+            <TouchableOpacity
+              accessibilityLabel="add-check-in"
+              onPress={() => {
+                addTodaysCheckIn(newStatus)
```
Comment (devinjameson) [inline src/MyHealth/SymptomLogContext.spec.tsx:35]: 🔥

File: src/MyHealth/SymptomLogContext.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+        expect(addCheckInSpy).toHaveBeenCalledWith({ ...newCheckIn })
+        expect(getByText(newStatus.toString())).toBeDefined()
+      })
+    })
+  })
+
+  describe("data seeding", () => {
+    it("fetch all daily check-in and log entries to serialize", async () => {
+      const checkIns = [{ date: Date.now, status: CheckInStatus.FeelingGood }]
+      const logEntries = [{ date: Date.now(), symptoms: [], id: "1" }]
```
Comment (devinjameson) [inline src/MyHealth/SymptomLogContext.spec.tsx:62]: Why is this `Date.now()` invoked but the one on like 61 is not?

File: src/MyHealth/SymptomLogContext.tsx [hunk 4]
```diff
@@ -8,50 +8,20 @@ import React, {
 
 import {
   SymptomLogEntry,
-  DailyCheckIn,
+  CheckIn,
   CheckInStatus,
   DayLogData,
-  serializeDailyLogData,
+  combineSymptomAndCheckInLogs,
```
Comment (devinjameson) [inline src/MyHealth/SymptomLogContext.tsx:14]: 👍

File: src/MyHealth/SymptomLogContext.tsx [hunk 5]
```diff
@@ -98,20 +81,23 @@ export const SymptomLogProvider: FunctionComponent = ({ children }) => {
...
 
-  const deleteLogEntry = async (deletedLogEntry: SymptomLogEntry) => {
-    const entriesWithoutTheDeletedLog = logEntries.filter((entry) => {
-      return entry.id !== deletedLogEntry.id
+  const addTodaysCheckIn = async (status: CheckInStatus) => {
+    const newCheckIn = { date: Date.now(), status }
+    await addCheckIn(newCheckIn)
+    setTodaysCheckIn(newCheckIn)
+    getCheckIns().then((checkIns) => {
+      setCheckIns(checkIns)
```
Comment (devinjameson) [inline src/MyHealth/SymptomLogContext.tsx:89]: Why am I thinking about 🐔 now.

**Review by devinjameson — APPROVED** [review]
:shipit: 🚀

File: src/MyHealth/SymptomLogContext.tsx (hunk 1 above)
Comment (aledustet) [reply src/MyHealth/SymptomLogContext.tsx:16]: We are moving the modules in a future iteration. 👍

**Pull request merged**